### PR TITLE
changed the api to naturally move us over from v1 to v2 input and structures

### PR DESCRIPTION
**Pull request by wheresrhys**
@matthew-andrews Added methods to get brand and primary concept prior to redoing cards

I also think it's better to have a single set of helpers that work with V1 and V2

### CODE REVIEW

**Comment by matthew-andrews** [comment]
> I also think it's better to have a single set of helpers that work with V1 and V2

please no >.<

much prefer to keep them separate so they can be disposed of easily……

if we need helpers for v2 then there's a problem…… it should be simple enough to not need helpers or we should request them to be simplified

**Comment by wheresrhys** [comment]
I think there's always gonna be some transforms we need to do on v2 e.g. attaching a http://next.ft.com/organistaion/ibm url to the IBM concept. 

How would keeping them separate make them easier to dispose of? Keeping them as one would make it easier to seamlessly update some property to use v2 over v1 as the api matures, without requiring a sweep of our apps to update all the 'methods from v1 to v2.

**Comment by matthew-andrews** [comment]
> I think there's always gonna be some transforms we need to do on v2 e.g. attaching a http://next.ft.com/organistaion/ibm url to the IBM concept.

My current plan is to use UUIDs for stream pages.

> How would keeping them separate make them easier to dispose of? Keeping them as one would make it easier to seamlessly update some property to use v2 over v1 as the api matures, without requiring a sweep of our apps to update all the 'methods from v1 to v2.

By not writing them in the first place?

**Comment by wheresrhys** [comment]
> My current plan is to use UUIDs for stream pages.

I think a separate url property is best all the same, even if it ends up just being a UUID in the end

This is all aimed at making article cards agnostic between capi1 and capi2 https://github.com/Financial-Times/next-article-card-component/blob/9f0a29e90a61e43064640f46a732a8d405180a0d/models/article-card.js but how primaryConcept is constructed also needs to be shared with the header, so pulling it out of article-card feels like the right thing to do.

**Comment by matthew-andrews** [comment]
@wheresrhys CAPI v2 currently has no concept of primary concept.

**Comment by matthew-andrews** [comment]
if we need primary concept in Next someone needs to tell the content team and get that prioritised………

**Comment by matthew-andrews** [comment]
re: url, isn't that what `webUrl` is for?

e.g. api.ft.com/content/47aa9aa4-a7a5-11e4-be63-00144feab7de

**Comment by wheresrhys** [comment]
- primaryConcept
This is the main concept we label an article with so it's as much a UX thing as it is content (maybe I've given it a misleading name and better to stick with primaryTag). It'd be, at the very least, premature to start asking UP to put in a property that is basically there to help us play around with navigation.

- url
yes that is what web url is for, but it'll be a long time until it points at next, so if we want to use it to populate links in next won't we need to transform it anyway? I'm not sure if capi2 even has weburls for things yet

**Comment by matthew-andrews** [comment]
that example i gave for the webUrl *was* from CAPI v2……

the logic for primaryConcept/Tag is a very special case.  i worry that it's overcomplicated and distracts from actually making the site…  but maybe we can have a module purely for this single purpose……

For the rest of it, can't the transform be a polyfill on top of V1 to make it V2-like instead so that when we ditch V1 we can just delete the file and the rest of the application needn't change?  I worry about the level of untangling this is going to require when we ditch it……

i should be able to throw a standard CAPI v2 article model at the article card and it should render……  the purpose of having transforms separate from the API client is that they ought not to be necessary much of the time…… so that they are optional.  e.g. grumman should never have to use this…

File: src/get-primary-image.js [hunk 1]
```diff
@@ -0,0 +1,19 @@
+'use strict';
+var tagToConcept = require('./util/tag-to-concept');
+
+module.exports = function (capi2, capi1) {
+	if (capi1.item.images) {
```
Comment (matthew-andrews) [inline src/get-primary-image.js]: we have main image in capi v2 now…… better if we integrated only with that and ignored v1 data

**Pull request closed without merging**